### DISCUSSION TITLE
Prototype of ERGSemantics_Essence for Thursday's Meeting

### DISCUSSION THREAD

**Discussion by EricZinda** [delph-in/docs#103] [Team Posts]
I don't know if I need to "@" people to get noticed in Github, so: @arademaker, @fcbond, @danflick, @olzama, @lmorgadodacosta:

I finished (most of) a [prototype of the ERG Semantics Essence page ](https://github.com/delph-in/docs/wiki/ERDWErgSemantics_Essence)for our Thursday meeting.  Original is [here](https://github.com/delph-in/docs/wiki/ErgSemantics_Essence).  Same goal as the reference page: The prototype attempts to describe the concept in a way that the target audience we discussed can understand. I also tried to make it work as it does now by keeping all the linguist-specific info in there. Let's see if we can meet both audience's needs.

This one is much longer and more different from the original than the Reference Page prototype.  I recommend reading it first and *then* reading the other to compare.  They are different enough in structure that it is tough to read them side by side. However, I'll claim that no information was lost (outside of a couple examples mentioned below), just presented differently.

I haven't gotten to the last few sections, but consider the ones that are there to be "done" as far as the prototype goes. I haven't bothered to add links to other information yet, but that should be added.

Same questions as the reference page:

Overall thoughts and comments?
Do you think it works for the target audience?
Do you think it succeeds in targeting both audiences? I.e. if we deleted the original, are we good with that?

More details if you want to know what changed:
- I tried to use less jargon and simplified the text throughout (except in the linguistic detail)
- To address both audiences, I tried a "callout" format for the deep linguistic descriptions. That way it is highlighted for the linguists and easy to skip for others.  Curious how you think that worked.
- The document has a relatively different structure than the original for flow and clarity reasons. I build a foundation of "what is MRS" and try to give you a working knowledge, then show how the ERG represents interesting examples in it.
- After the overview, the 3 sections that follow are really designed to help the user understand the MRS (more than the original), since it is the core interface and so they can follow the examples. I think it helps make the thing less scary and lets the reader feel like they really have their head around what this system produces.
- No information from the original was removed (outside of being rewritten sometimes) *except*: 
   - I used simpler (but more detailed) examples than "The cheerful children wanted to sing and dance" when introducing the notion of predicates (i.e. the yellow dog section)
   -   (5a) To apply this technique is impossible, (5b) Applying this technique is impossible and (6) *It is impossible applying this technique. were all removed since they didn't seem necessary , the points were made elsewhere. You can tell me if I'm wrong!

**Comment by arademaker** [comment]
I have many comments:

First, markup needs to be properly used. The quote was used to introduce the subsection in the first section? 

predicate-logic-like ?? What does that mean?  Functions, predicates and relations are quite different, the analogy with functions from a programming language is not correct. 
 
MRS stands for '**Minimal** Recursive Semantics' not 'Multiple Recursion Semantics (MRS)'. The explanation for this is given in the [Copestake's paper](https://www.cl.cam.ac.uk/~aac10/papers/mrs.pdf). 

The example of `compound` Is not useful  in this page, note that compound is another underspecified predication, this I learned from @guyemerson [here](https://github.com/delph-in/erg/issues/29).

**Comment by arademaker** [comment]
No no... you can't say:

> The predications in an MRS document have (often very blandly) named arguments such as ARG0 and ARG1.

See 19.3 from https://web.stanford.edu/~jurafsky/slp3/19.pdf.

**Comment by arademaker** [comment]
I am sorry, but there are many more problems... we need to find a better approach. Too many MRS, I don't think they are all necessary.  The format for presenting the scoped resolved MRS is not very practical.

**Comment by arademaker** [comment]
Finally, IF we are going to propose changes in the pages, we need to be more conservative and adopt some strategy to minimize unnecessary changes in the pages. So diff tools can be used to compare versions 

<img width="1812" alt="image" src="https://user-images.githubusercontent.com/86403/178761089-b943217e-bc89-4a78-b3de-305399fb47b8.png">

**Comment by olzama** [comment]
I think what we are learning from this is that are certain things which are important but only noticeable to specialists 😅 which makes sense. That why we have some of them on this team ;). 

Things like proper naming and formatting are of course very easily fixed...

**Comment by EricZinda** [comment]
Yeah, that's great! My objective for this prototype was to try to capture the overall feel of what a conceptual doc could look like that tried to hit all the audiences.  I love the detailed feedback, and we need to get there eventually, but mostly I was hoping to have this group read it, try to get past editing level stuff and decide:

Overall thoughts and comments?
Do you think it works for the target audience?
Do you think it succeeds in targeting both audiences? I.e. if we deleted the original, are we good with that?

I'll respond to @arademaker's individual comments, per comment

**Comment by EricZinda** [comment]
> I have many comments:
> 
> First, markup needs to be properly used. The quote was used to introduce the subsection in the first section?
> 
Not sure I understand. Which quote?

> predicate-logic-like ?? What does that mean? Functions, predicates and relations are quite different, the analogy with functions from a programming language is not correct.
> 
I understand what you mean. On the other hand, `foo(X, Y, Z)` sure looks like a function, and the implementation using it could literally have a python function named `foo(X, Y, Z)` that that is used when it is seen, so...Good discussion for later. 

> MRS stands for 'Minimal Recursive Semantics' not 'Multiple Recursion Semantics (MRS)'. The explanation for this is given in the Copestake's paper.
> 
Typo. Fixed.

> The example of compound Is not useful in this page, note that compound is another underspecified predication, this I learned from @guyemerson here.

That was copied from the original, so I wanted to leave it in. The point of it in the original, I think, was to highlight how the ERG not only has predications that match words in the original text, but also introduces purely semantic stuff (i.e. predications without a leading underscore) to richly tell you what was going on.

**Comment by EricZinda** [comment]
> No no... you can't say:
> 
> The predications in an MRS document have (often very blandly) named arguments such as ARG0 and ARG1.
> 
> See 19.3 from https://web.stanford.edu/~jurafsky/slp3/19.pdf.

That was (roughly) copied from the original text: "Because the bland semantic roles labels (ARG1, etc) are intended to have predicate specific interpretations".  Thanks for the doc reference, that is a good resource to potentially point to from the page. I'm unclear about our comfort level with linking and citations from"the literature" and what is considered definitive, etc. Be good to establish this (for me at least). Adding to issues list.

**Comment by EricZinda** [comment]
> I am sorry, but there are many more problems... we need to find a better approach. Too many MRS, I don't think they are all necessary. The format for presenting the scoped resolved MRS is not very practical.

It is actually less MRS than the original, but the original used a different form (whose name I forget) Simple MRS? Maybe that form reads better? We should discuss what canonical form we want for all the docs. Added to issues list.

I think of the MRS documents as being the "example code" of this world. It's the stuff you need to be able to read, want to see what gets generated, etc. It felt to me like part of the point of the beginning was to introduce people to the notion of the MRS, get to practice reading and understanding them, etc. I was focusing on "practical usage" of the MRS in all this.  Good discussion for the meeting tomorrow!

**Comment by EricZinda** [comment]
>  The format for presenting the scoped resolved MRS is not very practical.

@arademaker, could you say more about this? Do you mean the graph I have that shows it like this:
~~~
b) "Every dog is chasing the same cat"

         ┌_cat_n_1:x8
 _a_q:x8,h10,h11
             │            ┌_dog_n_1:x3
             └_every_q:x3,h5,h6
                             └_chase_v_1:e2,x3,x8
~~~
Or by "format" do you mean my approach to writing up how it works?

**Reply by arademaker** [reply]
> Do you mean the graph I have that shows it like this

Yes, I mean the graph representation. You are trying to provide one concrete scope of quantifiers among many alternatives from a given MRS.

**Comment by EricZinda** [comment]
> Finally, IF we are going to propose changes in the pages, we need to be more conservative and adopt some strategy to minimize unnecessary changes in the pages. So diff tools can be used to compare versions

It is a good discussion how conservative we need to be about changes. My view was: figure out what we want the docs to be and make them that way. Being able to diff with the original wasn't a priority. The point of this working group is to figure out how to get docs that address a larger target audience, so there could be some large changes to some of them. I'd hate to tie our hands in the approach.

All that said, this is prototype 1. I'm not claiming it shows what we go with, it is there for discussions (like this). Let's have a discussion about how much we care about restructuring the docs tomorrow.

**Comment by EricZinda** [comment]
BTW, @arademaker, I forgot to say up front: love the feedback! This is the kind of stuff we need to get into, for sure.

**Comment by arademaker** [comment]
> Do you mean the graph I have that shows it like this

Yes, I mean the ASCII graph. 

> about the format

I mean https://github.com/delph-in/docs/wiki/ERDWErgSemantics_Essence/_compare/1cbd8eebf02b8f8742679822b00ac28f8929bebb...a3d05614387798dd6fb9bcb035f653e44991130a

**Comment by EricZinda** [comment]
Got it. Yes, I'm open to better ideas for the format if you have them.

Re: the format, it was meant as an example of what they will see. I fixed it a little to maybe make it clearer.  It is now like this:

...

Throughout this document, attempts are made to give the linguistics background and reasoning behind certain behaviors, like this:

> Linguistics Background: The linguistics background would go here...

Understanding these explanations are not important for understanding and using the ERG, they are there to help those with a linguistics background to understand some of the subtleties of the grammar.

...

**Comment by guyemerson** [comment]
Somewhat belatedly, I'd like to summarise what I like about the ERDW_ImplicitLocatives page:

1. The brief beginner-friendly description
2. The "Counter examples" -- but this should be renamed, maybe "Negative examples" or "Related phenomena"
3. The detailed explanation of MRS

All of these can sit alongside what is currently on the Semantic Documentation pages, and would improve the pages.  A brief description can sit alongside the longer "Linguistic Characterization" section that is already there.  A verbose explanation of MRS can sit alongside the terse "ERS Fingerprint" that is already there.  Negative examples are useful for clarifying the scope of the page, and for helping someone find the right page if they're looking in the wrong place.

Adding these points to the existing documentation pages would enhance them, and I believe this would cover the intended use cases discussed at the summit.  It would be less work (both now and also when maintaining these pages in future) to have one set of documentation pages rather than splitting them into two.

**Comment by EricZinda** [comment]
Never too late, thanks for the feedback!
